### ISSUE TITLE
[Enchancement] Compare two XML files, ignoring element and attribute order

### ISSUE THREAD

**Issue by Forusim** [pnedev/compare-plugin#177]
Hello developers, it would very useful to have a XML-aware comparison.

Some applications produce XML files with different order of child items for same content, which makes text line comparison useless.

In the below example I would like to have an option to ignore item order.

```
<?xml version="1.0" encoding="UTF-8"?>
<foo>
     <bar1>stuff1</bar1>
     <bar2>stuff2</bar2>
</foo>
```

```
<?xml version="1.0" encoding="UTF-8"?>
<foo>
     <bar2>stuff2</bar2>
     <bar1>stuff1</bar1>
</foo>
```

I have found some solutions for this problem, but it would great to have this build into my favourite Notepad++.

https://stackoverflow.com/questions/40936205/intelligent-way-to-diff-xml
https://github.com/joh/xmldiffs

Hope more people will find this enhancement also useful.

**Comment by stiffsen** [comment]
I second this request. This would be a VERY useful feature. I have to diff xml configs so frequently and it is always a hassle.

Anyway, thanks for pointing out xmldiffs. I will use that for the time being.

**Comment by pnedev** [comment]
Hello all,

In a month a new plugin named ComparePlus will be out.
And eventhough it will not have explicit XML compare it has a command to Find Unique Lines between two documents that will be useful in the described case because it disregards lines order.
I will close this issue now. I would also add that it is out of the plugin scope to compare such specific cases. It is meant to be more generic comparison tool. Sorry about that.

BR

**Comment by stiffsen** [comment]
Thanks, I will try that.

However, as my demand for such a functionality grew stronger, and I even have to diff big collections of XMLs occasionally, I implemented a solution myself. I wrote a little python script that parses all XMLs files in a folder and saves them such that all nodes and properties go into a separate line and are alphabetically sorted. This way, I can use any diff tool, to check for differences. Especially useful is Total Commander's "Synchronize Dirs" feature, because that let's me diff the whole collection of XMLs in one go. :-)
